netmaster/objApi/apiController.go

```diff
@@ -1571,6 +1571,38 @@ func (ac *APIController) RuleCreate(rule *contivModel.Rule) error {
 		return core.Errorf("Policy not found")
 	}
 
+	if rule.Direction == "in" && rule.ToIpAddress != "" {
+		// rules from k8s network policy
+		// verify 'toIpAddress' is part of epg and subnet
+		if len(policy.LinkSets.EndpointGroups) <= 0 {
+			errMsg := fmt.Errorf("no endpoint groups linked to %s to use --to-ip-address", rule.PolicyName)
+			log.Error(errMsg)
+			return errMsg
+		}
+
+		for epgKey := range policy.LinkSets.EndpointGroups {
+			epg = contivModel.FindEndpointGroup(epgKey)
+			if epg == nil {
+				errMsg := fmt.Errorf("failed to find endpoint group %s to use --to-ip-address", epgKey)
+				log.Error(errMsg)
+				return errMsg
+			}
+			nwKey := epg.NetworkName
+			network := contivModel.FindNetwork(rule.TenantName + ":" + nwKey)
+			if network == nil {
+				errMsg := fmt.Errorf("failed to find network %s to use --to-ip-address", nwKey)
+				log.Error(errMsg)
+				return errMsg
+			}
+			if netutils.IsOverlappingSubnet(rule.ToIpAddress+"/32", network.Subnet) != true {
```

Comment:
Should we check for epg subnet also?

Reply:
it may not be required when using k8s network policy.
Because  epgs are created  without subnet.
But for other cases, I will add the check to avoid wrong config.